src/vscode-bicep/src/test/utils/time.ts

```diff
@@ -3,3 +3,20 @@
 export function sleep(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
+
+export async function retryWhile<T>(
+  func: () => Promise<T>,
+  predicate: (result: T) => boolean,
+  retryOptions = {
+    interval: 2000,
+    count: 3,
+  }
+): Promise<T> {
+  let result = await func();
+
+  while (predicate(result) && retryOptions.count--) {
```

Comment:
Will the change to `count` propagate outside of the function call? If yes, then we should probably make a copy first. Otherwise, you could get really weird behavior if you try to reuse retry options across function calls.

Reply:
Oops...never intended to write code like that. Pushed out a new commit to make that parameter read-only.